File: examples/cmp/cpp/correct.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // Maximum Subarray Problem

    int n; cin >> n;
    vector<int> values(n);
    for(int &a: values)
        cin >> a;
    
    int best = 0, sum = 0;
    
    // i+1 was added intensively to generate an error
    for (int i = 0; i < n; i++) {
        sum = max(values[i], sum + values[i]);
        best = max(best, sum);
    }
    cout << best << "\n";
    
    return 0;
}

File: examples/cmp/kotlin/correct.kt
fun main() {
    // Maximum Subarray Problem

    val n = readLine()!!.toInt()
    val a = readLine()!!.split(" ").map { it.toInt() }

    var best = 0
    
    // n-2 was added intensively to generate an error
    for (i in 0..n-1) {
        var sum = 0
        for (j in i..n-1) {
            sum += a[j];
            best = best.coerceAtLeast(sum);
        }
    }
    println(best)
}